### PR TITLE
feat(agents): add feature delivery manager

### LINKED ISSUE #370: Rebaseline the build-engine agent system against the current agent surface
## Summary

Build a **Feature Delivery Manager** system that takes one already-accepted issue through implementation, review, testing, QA, PR publication, and an explicitly approved deployment. It is the delivery counterpart to #369 / #371: Backlog Maintainer owns backlog decisions and issue-write approval; Feature Delivery Manager delivers accepted scope only. It must use the actual target agent surface, not assume absent agents or capabilities.

## Design of record — first

Before implementing any `.github/agents/*.agent.md`, write and review `docs/agents/feature-delivery-manager.md` as the design of record. Agent files derive from it and reference shared skills/specs rather than duplicating conventions or deployment commands.

The design must contain a **verified platform-capability table** for the target surface: session branch/ref creation semantics; session messages; child-session worktree isolation; live GitHub MCP tool names; generic execution limitations; and credential boundaries. Each entry records evidence, status, and fallback. No unverified capability may be stated as fact. A missing or differently named required tool is a blocking configuration mismatch, never grounds to widen permissions with `github/*`.

It must also inventory `.github/agents/`, `.github/skills/`, applicable instructions/specs, custom-agent tools, and live GitHub MCP read/write names before finalizing frontmatter. Before creating any scoped instruction file, it must inventory the custom-instruction format and the supported `applyTo` matching semantics on every target agent surface; candidate glob syntax is not evidence.

### Execution-boundary verification

Generic execute/shell access can run arbitrary Git, GitHub, or deployment commands. Tool allowlists alone therefore **cannot** make execution-bearing roles structurally unable to mutate, publish, or deploy. The design must inventory whether every target surface provides scoped/sandboxed command execution and isolated credentials. If it does not, the no-edit/no-publish/no-deploy rules below are mandatory behavioral controls: use fresh isolated child branches/worktrees; omit GitHub MCP write tools; avoid credential injection; log exact commands; and report residual risk. This is a blocker to claiming structural enforcement, **not** to operating the system with documented behavioral controls.

## Scoped instruction map

The existing `.github/instructions/` pattern is intentionally retained and improved: it currently provides narrow guidance for article Markdown, React components, and `*.test.tsx`; this work must extend that pattern, not replace it with a monolith.

### Layering and selection

- Root `.github/copilot-instructions.md` remains repository orientation. `docs/specs/` remains authoritative for architecture, quality, testing, accessibility, non-goals, and related product rules. Scoped instruction files contain only local, actionable conventions and link upward; they must not copy whole specs or delivery-system policy.
- Before creating files, record the verified custom-instruction schema/frontmatter and supported `applyTo` match semantics for each target agent surface. Name files and use patterns only after that evidence; if a target cannot express a required scope, document its safe fallback rather than assuming glob behavior.
- The Delivery Brief selects instruction files strictly by modified-path relevance and records both selected files and explicit exclusions. Example: a React page/route/SSR change selects the verified React-page scope (and applicable component/test scope), while a React-only task neither loads Terraform/Azure/workflow rules nor triggers those specialists. A change spanning scopes selects only the corresponding verified files.

### Intended map after verification

Create a small, layered set of appropriately named `*.instructions.md` files covering: React pages/routing/SSR/prerender and build tooling; core/content/data access; all TypeScript test files (including `.test.ts`, improving the present `.test.tsx`-only coverage); GitHub Actions workflows; Terraform infrastructure; Azure Static Web Apps/deployment configuration; custom-agent definitions; and skill/design documentation. Preserve and refine the existing article Markdown and React-component scopes. Avoid overlapping catch-all files where a local scope is sufficient.

GitHub Actions, Terraform, and Azure scopes must require least privilege, no tracked secrets, environment separation, reviewable and idempotent infrastructure/deployment changes, and existing project conventions. They must prohibit new credentials, hardcoded resource names, and unrelated deployment paths.

The custom-agent-definition and skill/design-documentation scopes must require least privilege; verified tools and frontmatter; explicit artifact contracts; session/worktree/approval rules; and avoiding duplicated judgment already owned by shared skills or specs. Put the shared delivery system in its design of record, not repetitively in every agent file.

### Instruction-loading validation

Add a validation fixture/check that demonstrates each intended scope loads its expected instruction file and that unrelated scopes do not load it. Use only existing project capabilities; if the host lacks an automatable assertion, document a repeatable manual verification (target surface, representative paths, observed loaded files, negative cases, date/version) and make it a release-blocking verification for this work.
## Roles and boundaries

| Role | Owns | Must not do | Boundary |
| --- | --- | --- | --- |
| **Feature Delivery Manager** | Routing accepted scope, choosing triggered specialists, tracking SHA-bound artifacts/sessions, approval gates, outcome | Edit code, build, publish, deploy, or make backlog decisions | Coordination, read/search, GitHub read-only |
| **Developer** | Approved implementation in its live mutable branch/worktree; focused commits | Publish, deploy, alter scope, silently waive findings | read/search/edit/execute; no GitHub write/deployment tools; behavioral no-publish/no-deploy control if execution is unscoped |
| **Code Reviewer** | Review exact snapshot against issue, applicable scoped instructions, and specs | Edit, mutate, publish, deploy | read/search only |
| **Test Engineer** | Deterministic checks and raw results, including applicable scoped instructions | Edit, treat failure as acceptable, publish, deploy | read/search/execute; behavioral controls apply where execution is unscoped |
| **QA Engineer** | Observable journeys, responsive/browser/a11y/theme/reduced-motion/SSR-prerender checks as relevant, against applicable scoped instructions and specs | Edit, downgrade defects, publish, deploy | read/search/execute/browser; behavioral controls apply where execution is unscoped |
| **Release Manager** | Create/update the explicitly approved PR for one verified SHA | Edit code, deploy, merge, change approved scope/payload | requested GitHub PR write only after independently validating Approval Record; any generic execution uses behavioral controls |
| **Deployment Manager** | Deploy one approved SHA by named mechanism to named Azure environment; report URL/status | Edit code, create/merge PR, substitute SHA/environment/mechanism | deployment execution only after independently validating Approval Record; structural scope only if verified, otherwise behavioral controls and residual risk |

Where a role has execution, the design must state whether the boundary is structural or behavioral and record residual risk; it must never claim structural inability without the verified capability evidence.

### On-demand specialists

Keep these seven read-only advisors: **React, Frontend, Accessibility, GitHub Actions, Terraform, Azure, Security**. They are strictly on-demand, selected only when explicit trigger/path/risk criteria in the Delivery Brief match (respectively: React/SSR; UI/layout; WCAG; `.github/workflows/**`; `infra/**`; Azure/SWA; secrets/permissions/dependencies/input). There is no mandatory seven-agent hop. Their **Specialist Guidance** is advisory, must consider the selected scoped instructions plus specs, and cannot alter code or promote a build.

Developer resolves implementation-level recommendations within the accepted Delivery Brief; Reviewer checks that choice against the applicable scoped instructions and specs. Conflicting guidance that changes scope, security posture, user behavior, or delivery risk becomes a **Decision Request** to the human through Feature Delivery Manager.

## Session, worktree, and communication contract

Tracked child sessions are required where supported; otherwise stop with a complete Delivery Brief and name the next manual role. One-off subagents are optional, short read-only specialist consultations only; they are not substitutes for durable branch-isolated, follow-up-capable delivery phases.

1. Feature Delivery Manager creates one Developer child session/worktree per issue. Developer alone owns its live mutable branch/worktree; all implementation/remediation stays there.
2. Developer emits an **Implementation Receipt** only after committing. Its branch and receipt SHA—not uncommitted files—are the hand-off.
3. For **each** Code Reviewer, Test Engineer, and QA Engineer phase, the manager creates a **new** child session on a **distinct child branch/worktree**, whose start commit is created from the Developer branch at the Implementation Receipt SHA. Before phase work starts, the manager verifies that child's checked-out HEAD equals that SHA and records the verification.
4. Never tell another worktree to check out Developer's live branch; never use uncommitted files as input; never assume a detached-SHA checkout is supported. If verified platform semantics cannot create the required child branch/worktree snapshot, stop that automated phase and use the documented manual fallback.
5. A new Developer commit invalidates every prior snapshot phase and receipt. Recreate Review, Test, and QA sessions from the new receipt SHA; do not reuse green results.
6. Feature Delivery Manager alone routes messages, creates tracked sessions with `coordinate_with_creator: true` when supported, records session IDs in the delivery ledger, waits for terminal results, and sends complete artifacts. Children do not direct siblings, create replacements, or rely on shared memory.
7. Read-only specialists may work from the accepted base revision only when their guidance needs no implementation state.

## Subsession-first orchestration

On surfaces verified to provide `create_session`, `get_session`, and `send_session_message`, delivery is subsession-first. A subagent is not an equivalent substitute for a stateful delivery phase because it does not establish the required durable, visible, branch-isolated, follow-up-capable workspace.

- Feature Delivery Manager creates tracked child subsessions for Developer, every Code Reviewer/Test Engineer/QA Engineer snapshot phase, Release Manager, and Deployment Manager, records their IDs, and uses `coordinate_with_creator: true` where verified.
- Developer owns the one live mutable delivery branch/worktree. Every Reviewer, Test Engineer, and QA Engineer gets a **fresh** subsession on its own **distinct child branch/worktree**, starting from the Developer branch at the current Implementation Receipt SHA; the manager verifies and records that its checked-out HEAD is that SHA before work begins.
- A Developer commit invalidates all earlier Review/Test/QA subsessions and receipts. The manager creates replacements from the new receipt SHA; it never directs a child to check out the Developer live branch, uses uncommitted state, or assumes detached-SHA checkout support.
- Release and Deployment each receive a distinct subsession only after their respective Approval Record is valid. The manager supplies the complete artifact at creation and uses `send_session_message` only to resume the same child with a decision or remediation request.
- Children do not direct siblings, silently create replacements, or depend on shared conversational memory. Optional subagents remain short, read-only specialist consultation only and never own a worktree, approval, publication, or deployment state.
- Without verified tracked-subsessions/worktree semantics, the automated cycle is unavailable: stop with the complete Delivery Brief and next manual role; do not claim one-off agents provide equivalent guarantees.

## Delivery flow

```mermaid
flowchart TD
    H[Accepted issue] --> M[Feature Delivery Manager]
    M --> S[Triggered specialists: read-only guidance]
    S --> D[Developer: live mutable worktree]
    D --> C[Committed Implementation Receipt SHA]
    C --> R[New SHA-verified Reviewer snapshot]
    R -->|findings| D
    R -->|pass| T[New SHA-verified Test snapshot]
    T -->|failure| D
    T -->|pass| Q[New SHA-verified QA snapshot]
    Q -->|pass for same SHA| G1{Human PR decision}
    G1 --> P[Release Manager]
    P --> G2{Human deployment decision}
    G2 --> X[Deployment Manager]
```

A blocked required tool, failed validation, or unresolved high-confidence finding stops publication; report the blocker, never skip it. Reviewer and QA may find issues beyond specialist scope.

## Approval gates and human control

At both PR and deployment gates, the human may **approve, edit, defer, reject, or cancel**:

- **approve:** creates an Approval Record for exactly the proposed action.
- **edit:** invalidates the proposal and returns it to the correct preparatory phase.
- **defer:** pauses with no retry.
- **reject/cancel:** terminates that requested action; there is no automatic alternative, retry, or target substitution.

Any approval is invalidated by a new commit; changed PR title, body, base, or head; changed environment or mechanism; or a scope change. Release Manager and Deployment Manager must not silently retry.

Feature Delivery Manager presents PR repository, source branch/SHA, exact title/body/base/head, and verification receipts. Deployment requires a separate approval naming repository, SHA, environment (`preview` or `production`), and mechanism; PR approval is never deployment approval. Neither manager may merge, change environment, replace SHA, or inject deployment tokens into repository files/artifacts.

**Release Manager and Deployment Manager each independently require a valid structured Approval Record, exactly as issue-writer does; Feature Delivery Manager cannot be the sole enforcement point.** They refuse to act without it.

## Artifact contracts

Every artifact has `delivery_id`, issue URL/number, source branch, source commit SHA, input references, status (`pass`, `fail`, `blocked`, `needs-approval`), timestamp, and session ID where applicable.

- **Delivery Brief:** accepted issue, objective, constraints, in/out scope, base branch/SHA, modified-path relevance assessment, selected scoped-instruction files (and explicit exclusions), explicit specialist trigger/path/risk criteria, checks, publication/deployment intent.
- **Specialist Guidance:** role, facts/citations, constraints, recommendations, verification, blockers, source SHA.
- **Decision Request:** conflicting guidance, impact on scope/security/user behavior/delivery risk, options, and human decision needed.
- **Implementation Receipt:** paths, commit SHA/branch, tests, limitations, referenced Brief/Guidance.
- **Review Verdict / Test Receipt / QA Verdict:** source SHA, status, evidence (findings or exact commands/output or routes/viewports/themes/input methods/evidence), and remediation when needed.
- **Approval Record:** exact action; repository; source branch/SHA; exact PR payload **or** named environment and deployment mechanism; expiration/invalidation conditions; and the human's verbatim approval quote or unambiguous approval-message reference.
- **Release Proposal/Receipt / Deployment Proposal/Receipt:** Approval Record reference, one SHA, approved payload/environment/mechanism, and resulting PR URL/number or deployment URL/status.

## Acceptance criteria

- [ ] Design of record is written and reviewed **before** any `.github/agents/*.agent.md` implementation; it contains the verified capability table, roles, boundary classification, artifacts, gates, failure routing, session semantics, credential boundaries, and manual fallbacks.
- [ ] Live agents/skills/tools/MCP names are inventoried; custom-instruction format and supported `applyTo` semantics are verified on each target surface before instruction files are created; unverified capability is never asserted and missing required tooling blocks structural-enforcement claims.
- [ ] Execution-boundary verification documents scoped execution/credential evidence or mandatory behavioral controls, exact-command logging, and residual risk for execution-bearing roles.
- [ ] Create user-invocable Feature Delivery Manager, Developer, Code Reviewer, Test Engineer, QA Engineer, Release Manager, Deployment Manager, and seven strictly on-demand specialist agents.
- [ ] Reuse `react-app`, `validate-app`, `swa-deploy`, `triage-accessibility`, path instructions, and `docs/specs/` without duplicating them.
- [ ] Preserve and improve the current narrow article Markdown, React-component, and `*.test.tsx` instruction pattern with a verified, layered scoped map; keep root orientation and specs authoritative, and keep local instructions actionable and non-duplicative.
- [ ] Add the verified scopes for React pages/routing/SSR/prerender and build tooling, core/content/data access, all TypeScript tests including `.test.ts`, GitHub Actions, Terraform, Azure Static Web Apps/deployment configuration, custom-agent definitions, and skill/design documentation; the infrastructure/deployment scopes enforce least privilege, no tracked secrets, environment separation, idempotence, and no credential/resource/deploy-path expansion.
- [ ] Delivery Brief selection uses modified-path relevance, records selected and excluded scopes, and avoids unrelated rules/specialists.
- [ ] Validate instruction loading with an existing-capability fixture/check for all intended positive and negative scopes, or a documented repeatable manual verification when automation is unavailable.
- [ ] Developer is the only code mutator; Reviewer and specialists are read-only. Test/QA report evidence without editing; Release/Deployment act only within independently validated Approval Records.
- [ ] Code Reviewer, Test Engineer, QA Engineer, and every triggered Specialist explicitly assess the scoped instructions selected in the Delivery Brief as well as the applicable specs.
- [ ] Enforce SHA-bound artifacts and snapshot protocol: Developer live ownership, committed-only inputs, a newly created distinct branch/worktree and verified matching HEAD for every Review/Test/QA phase, and phase recreation after every new Developer commit.
- [ ] Implement manager-routed tracked-session coordination plus documented manual fallback; never imply shared uncommitted state, live-branch checkout by another worktree, or unsupported detached-SHA behavior.
- [ ] Implement approve/edit/defer/reject/cancel, all invalidation rules, refusal checks, and no-silent-retry behavior at both gates.
- [ ] Exercise a suitable real issue through guidance where triggered, implementation, review, deterministic testing, QA, PR/deployment approval-refusal checks, and final report. Do not deploy production merely to validate the system.
- [ ] Do not reference `.github/agents/react-app.agent.md` unless this scope explicitly creates it.

## Likely affected files

- `docs/agents/feature-delivery-manager.md` (new)
- `.github/agents/feature-delivery-manager.agent.md` and role/specialist `.agent.md` files (new)
- Existing `.github/instructions/articles.instructions.md`, component guidance, and `*.test.tsx` guidance (preserved and improved, not replaced)
- New, appropriately named scoped `*.instructions.md` files under `.github/instructions/` only after `applyTo` verification: React pages/routing/SSR/prerender and build tooling; core/content/data access; all TypeScript tests (including `.test.ts`); GitHub Actions workflows; Terraform; Azure Static Web Apps/deployment configuration; custom-agent definitions; and skill/design documentation
- A validation fixture/check or documented manual-verification record for instruction loading
- Existing shared skills/instructions only for minimal, non-duplicative hand-off references

## Related

- #369 — backlog-maintainer agent system
- #371 — backlog-maintainer design of record
- #363 — articles agent system rebaseline

### PR DESCRIPTION
**Pull request by lfarci**
## Why

The project needed a delivery workflow that can carry accepted issues through implementation, evidence gathering, and human-controlled publication without assuming unavailable platform permissions.

## Approach

- Add a capability-verified Feature Delivery Manager design with SHA-bound artifacts, scoped instruction selection, approval gates, and documented manual fallbacks.
- Add Developer, Review, Test, QA, Release, Deployment, and on-demand specialist agent profiles with least-privilege boundaries.
- Extend layered path instructions across app, content, workflow, infrastructure, Azure, and agent-system files, with a Node fixture for intended instruction matching.

## Important constraints

Automated PR publication and deployment intentionally remain blocked until the live GitHub MCP tool names, branch-at-SHA session creation, and executing credential authorization are independently verified. The design records the safe manual fallback rather than widening permissions.

### CODE REVIEW

File: .github/agents/specialist-accessibility.agent.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+tools: ["read", "search"]
+user-invocable: true
+---
+
+# Accessibility Specialist
+
+Provide Specialist Guidance only when the Delivery Brief has an accessibility trigger.
+Assess selected instructions, `triage-accessibility`, and `docs/specs/accessibility.md`.
+Cite affected criteria, facts, recommendations, verification, and blockers for the
+source SHA. Do not edit, execute, publish, deploy, or change accepted scope.
```
[the remark this answers is not in the document]
Comment (lfarci) [reply .github/agents/specialist-accessibility.agent.md:17]: Accepted. All eight specialist profiles, including Debugging, now require the shared Specialist Guidance artifact envelope from the delivery-manager design.
